### PR TITLE
feat: Add new Saddle pools to 4 chains, plus new tokens.

### PR DESCRIPTION
**Pull request by hammeiam**
# Description

- Add new Saddle pools to mainnet
- Create Saddle pool definitions for Fantom, Arbitrum, and Optimism
- Add new tokens to chains where needed

# Testing Instructions

n/a

# Checklist

-   [x] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [x] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

**Review by kyu-c — COMMENTED** [review]
LGTM overall!

Left some comments about dry pools. 
Once those are removed, I can do some testing on my end.

File: src/asset-swapper/utils/market_operation_utils/constants.ts [hunk 1]
```diff
@@ -1636,46 +1686,149 @@ export const XSIGMA_MAINNET_INFOS: { [name: string]: CurveInfo } = {
...
+        pool: SADDLE_MAINNET_POOLS['4Pool'],
         gasSchedule: 150e3,
-    },
+    }),
+    [SADDLE_MAINNET_POOLS.fraxBP]: createSaddleSwapPool({
+        tokens: [MAINNET_TOKENS.USDC, MAINNET_TOKENS.FRAX],
+        pool: SADDLE_MAINNET_POOLS.fraxBP,
+        gasSchedule: 150e3,
+    }),
+    [SADDLE_MAINNET_POOLS.frax3Pool]: createSaddleSwapPool({
```
[the remark this answers is not in the document]
Comment (hammeiam) [reply src/asset-swapper/utils/market_operation_utils/constants.ts]: Done!

**Comment by hammeiam** [comment]
@kyu-c made the changes you requested and merged in the latest master 👍

**Comment by kyu-c** [comment]
Thanks @hammeiam !! 

I have done testing and it looks good. For Arbitrum though there were some issues with testing (we haven't launched on Arbitrum). My recommendation is to merge the changes for Ethereum and Optimism first and revisit Arbitrum. 

Looks like there is an issue with our CI for an external PR. 

For the interest of time, as it might take a while to fix the CI issue, I created a new PR (#964) with small modifications.

1) Adjusted gas schedule a bit (was underestimating for a few pools)
2) Removed some dead code (e.g. Fantom related from the very first commit)
3) Ran formatter
4) Temporarily removed Arbitrum support (left the metadata there so that it's easier to re-enable in the future)

If you don't mind, I can merge #964 PR instead. If you prefer you can also get the diffs from #964 and merge this PR! Let me know what you prefer. :)

**Comment by hammeiam** [comment]
> Thanks @hammeiam !!
> 
> I have done testing and it looks good. For Arbitrum though there were some issues with testing (we haven't launched on Arbitrum). My recommendation is to merge the changes for Ethereum and Optimism first and revisit Arbitrum.
> 
> Looks like there is an issue with our CI for an external PR.
> 
> For the interest of time, as it might take a while to fix the CI issue, I created a new PR (#964) with small modifications.
> 
>     1. Adjusted gas schedule a bit (was underestimating for a few pools)
> 
>     2. Removed some dead code (e.g. Fantom related from the very first commit)
> 
>     3. Ran formatter
> 
>     4. Temporarily removed Arbitrum support (left the metadata there so that it's easier to re-enable in the future)
> 
> 
> If you don't mind, I can merge #964 PR instead. If you prefer you can also get the diffs from #964 and merge this PR! Let me know what you prefer. :)

@kyu-c That works for me, feel free to pull the trigger!

**Comment by kyu-c** [comment]
@hammeiam great! I should be able to merge soon. 

Thanks for the contribution! ❤️

**Pull request closed without merging**